config/types/systemd.go

```diff
@@ -40,6 +44,13 @@ type SystemdUnitDropIn struct {
 func init() {
 	register2_0(func(in Config, ast validate.AstNode, out ignTypes.Config, platform string) (ignTypes.Config, report.Report, validate.AstNode) {
 		for _, unit := range in.Systemd.Units {
+			if templating.HasTemplating([]string{unit.Contents}) {
+				contents, err := templating.PerformTemplating(platform, []string{unit.Contents})
+				if err != nil {
+					return out, report.Report{}, ast
+				}
+				unit.Contents = strings.Join(contents, "\n")
+			}
```

Comment:
If templating is used, the units must also have:
```
Requires=coreos-metadata.service
After=coreos-metadata.service
EnvironmentFile=/run/metadata/coreos
```

Here's where these lines are added for the generated units:
https://github.com/coreos/container-linux-config-transpiler/blob/2458180e0884a43c485d0980f8db62d171a40dd4/config/types/common.go#L79

Unfortunately users may have added these lines themselves to their units, so you should probably check for those. Maybe modify [`types/util/UnitSection.Add`](https://github.com/coreos/container-linux-config-transpiler/blob/2458180e0884a43c485d0980f8db62d171a40dd4/config/types/util/unit.go#L33) to check for preexisting matching keys, and to blow them away before appending the new value.